File: ws01_plubming/src/py04_param/py04_param/demo0_param.py
"""
需求：订阅发布方发布的消息，并输出到终端
步骤：
1.导入模块
2.初始化 ROS2 客户端
3.定义节点类
4.调用 spin 函数
5.释放资源
"""

import rclpy
from rclpy.node import Node


#自定义节点类
class MyParam(Node):
    def __init__(self):
        super().__init__('myparam_node_py')
        self.get_logger().info('参数API使用')
        # 创建参数对象
        p1 = rclpy.Parameter("car_name",value="Tiger")
        p2 = rclpy.Parameter("widht",value=1.5)
        p3 = rclpy.Parameter("wheels",value=2)

        # 解析参数
        self.get_logger().info("car_name=%s"%p1.value)
        self.get_logger().info("widht=%s"%p2.value)
        self.get_logger().info("wheels=%s"%p3.value)

        self.get_logger().info("key = %s"%p1.name)

def main(args=None):
    #初始化ROS2客户端
    rclpy.init()
    #调用spin函数 并传入节点对象
    node = MyParam()
    rclpy.spin(node)
    #资源释放
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ws01_plubming/src/cpp04_param/src/demo0_param.cpp
/*
需求：演示api实现
创建
  ros2 pkg create cpp04_param --build-type ament_cmake --dependencies rclcpp --node-name demo0_param
步骤：
1.包含头文件
2.初始化 ROS2 客户端
3.定义节点类

4.调用 spin 函数，并传入节点对象指针
5.释放资源
*/
#include "rclcpp/rclcpp.hpp"



// 自定义节点类
class MyParam: public rclcpp::Node{
public:
    MyParam():Node("myparam_node_cpp"){
      RCLCPP_INFO(this->get_logger(),"参数API使用");
      // 参数对象创建
      rclcpp::Parameter p1("car_name","tiger");
      rclcpp::Parameter p2("height",1.68);
      rclcpp::Parameter p3("wheels",4);
      // 解析值
      RCLCPP_INFO(this->get_logger(),"car_name = %s",p1.as_string().c_str());
      RCLCPP_INFO(this->get_logger(),"height = %.2f",p2.as_double());
      RCLCPP_INFO(this->get_logger(),"wheels = %ld",p3.as_int());

      // 获取参数键
      RCLCPP_INFO(this->get_logger(),"name = %s",p1.as_string().c_str());
      RCLCPP_INFO(this->get_logger(),"type = %s",p1.get_type_name().c_str());
      RCLCPP_INFO(this->get_logger(),"value2string = %s",p2.value_to_string().c_str());

      rclcpp::ParameterValue value();

    }
};

int main(int argc, char const *argv[])
{
    // 初始化ROS2客户端 
    rclcpp::init(argc,argv);
    // 调用spin函数 传入对象指针
    rclcpp::spin(std::make_shared<MyParam>());
    // 资源释放
    rclcpp::shutdown();
    return 0;
}